src/pages/HomePage/components/ProductListSection.tsx

```diff
@@ -1,86 +1,167 @@
-                            <ProductItem.Rating rating={3} />
-                            <ProductItem.Price>5.00</ProductItem.Price>
-                        </ProductItem.MetaLeft>
-                        <ProductItem.FreeTag type="gluten" />
-                    </ProductItem.Meta>
-                    <Counter.Root>
-                        <Counter.Minus onClick={() => {}} disabled={true} />
-                        <Counter.Display value={3} />
-                        <Counter.Plus onClick={() => {}} />
-                    </Counter.Root>
-                </ProductItem.Root>
-
-                <ProductItem.Root onClick={() => handleClickProduct(3)}>
-                    <ProductItem.Image src="/moon-cheese-images/tea-1-1.jpg" alt="문라이트 카모마일 티" />
-                    <ProductItem.Info title="문라이트 카모마일 티" description="달빛 같은 부드러운 허브차" />
-                    <ProductItem.Meta>
-                        <ProductItem.MetaLeft>
-                            <ProductItem.Rating rating={5} />
-                            <ProductItem.Price>$7.00</ProductItem.Price>
-                        </ProductItem.MetaLeft>
-                        <ProductItem.FreeTag type="caffeine" />
-                    </ProductItem.Meta>
-                    <Counter.Root>
-                        <Counter.Minus onClick={() => {}} disabled={true} />
-                        <Counter.Display value={3} />
-                        <Counter.Plus onClick={() => {}} />
-                    </Counter.Root>
-                </ProductItem.Root>
-            </Grid>
-        </styled.section>
-    );
+import { Counter, SubGNB, Text } from '@/ui-lib';
+import { useState } from 'react';
+import { useNavigate } from 'react-router';
+import { Box, Grid, styled } from 'styled-system/jsx';
+import ProductItem from '../components/ProductItem';
+import { useSuspenseQuery, useQueryClient } from '@tanstack/react-query';
+import { productListQueryOptions, queryKeys as productQueryKeys } from '@/apis/product';
+import { exchangeRateQueryOptions } from '@/apis/exchange';
+import { groupBy } from 'es-toolkit';
+import type { Product, ProductCategory } from '@/apis/types';
+import { formatPrice } from '@/utils/price';
+import { useAtom } from 'jotai';
+import { cartItemsAtom } from '@/atoms/cart';
+import { type CurrencyType } from '@/atoms/currency';
+import DataWrapper from '@/components/DataWrapper';
+
+type ProductCategoryOptions = ProductCategory | 'ALL';
+
+interface AddCounterProps {
+  onMinus: () => void;
+  onPlus: () => void;
+  stock: number;
+  amount: number;
+}
+
+function AddCounter({ onMinus, onPlus, stock, amount }: AddCounterProps) {
+  const isMaxAmount = amount >= stock;
+
+  return (
+    <Counter.Root>
+      <Counter.Minus onClick={onMinus} disabled={amount === 0} />
+      <Counter.Display value={amount} />
+      <Counter.Plus onClick={onPlus} disabled={isMaxAmount} />
+    </Counter.Root>
+  );
+}
+
+interface Props {
```

Comment:
```suggestion
interface ProductProps {
```

AddCounterProps로 이름 지어주신 것 처럼, 이 부분도 컴포넌트이름+Props 로 일관성 있으면 좋을 것 같아요